app/controllers/api_controller.rb

```diff
@@ -88,6 +89,12 @@ def authorize_with_headers
     end
   end
 
+  def set_headers
+    response.headers['Content-Type'] = 'text/json'
+    response.headers['Content-disposition'] = "attachment;\
+    filename='" + params['controller'].to_s + "'.json"
```

Comment:
try to avoid using params for static values.
I would rather use request.env['PATH_INFO']

Reply:
add this header checking at least to one controller spec